usaspending_api/recipient/v2/views.py

```diff
@@ -1,94 +1,168 @@
-        elif not year or year > latest or year == 'all' or year == 'latest':
-            return state_data[latest]
-        else:
-            raise InvalidParameterException('Invalid year: {}.'.format(year))
+def populate_fips():
+    global VALID_FIPS
 
-    def get(self, request, fips):
-        get_request = request.query_params
-        year = get_request.get('year')
+    if not VALID_FIPS:
+        VALID_FIPS = {fips_code: {'code': state_code, 'name': state_name} for fips_code, state_code, state_name
+                      in list(StateData.objects.distinct('fips').values_list('fips', 'code', 'name'))}
 
-        fips = fips.zfill(2)
-        state_data_qs = StateData.objects.filter(fips=fips)
-        if not state_data_qs.count():
-            raise InvalidParameterException('Invalid FIPS ({}) or data unavailable.'.format(fips))
 
-        state_data_results = state_data_qs.values()
-        general_state_data = state_data_results[0]
+def validate_fips(fips):
+    global VALID_FIPS
+    populate_fips()
+
+    if fips not in VALID_FIPS:
+        raise InvalidParameterException('Invalid fips: {}.'.format(fips))
+
+    return fips
+
+
+def validate_year(year=None):
+    if year and not (year.isdigit() or year in ['all', 'latest']):
+        raise InvalidParameterException('Invalid year: {}.'.format(year))
+    return year
 
-        # recreate filters
-        filters = {'place_of_performance_locations': [{'country': 'USA', 'state': general_state_data['code']}]}
+
+def reshape_filters(state_code=None, year=None, award_type_codes=None):
+    # recreate filters
+    filters = {}
+
+    if state_code:
+        filters['place_of_performance_locations'] = [{'country': 'USA', 'state': state_code}]
+
+    if year:
         today = datetime.now()
         if year and year.isdigit():
             time_period = [{
-                'start_date': '{}-10-01'.format(int(year)-1),
+                'start_date': '{}-10-01'.format(int(year) - 1),
                 'end_date': '{}-09-30'.format(year)
             }]
         elif year == 'all':
             time_period = [{
-                'start_date': '2008-10-01',
+                'start_date': settings.API_SEARCH_MIN_DATE,
                 'end_date': datetime.strftime(today, '%Y-%m-%d')
             }]
-        elif year == 'latest' or not year:
+        else:
             last_year = today - relativedelta(years=1)
             time_period = [{
                 'start_date': datetime.strftime(last_year, '%Y-%m-%d'),
                 'end_date': datetime.strftime(today, '%Y-%m-%d')
             }]
+        filters['time_period'] = time_period
+
+    if award_type_codes:
+        filters['award_type_codes'] = award_type_codes
+
+    return filters
+
+
+def obtain_state_totals(fips, year=None, award_type_codes=None, subawards=False):
+    filters = reshape_filters(state_code=VALID_FIPS[fips]['code'], year=year, award_type_codes=award_type_codes)
+
+    if subawards:
+        pass
```

Comment:
Is the plan to support subawards in the future? 

If we're skipping this every time, is it better to remove this `if` block and replace the else below with `if not subawards`?

Reply:
Subawards is outside the scope of this PR. I know subawards are expected to be added in the future but I do not know the specific requirements. @dpb-bah can add to the discussion

Reply:
Subawards was in the original design and then removed with the possibility it may come back. For the time being we can do the `if not subawards`